commands/keplr.js

```diff
@@ -238,12 +267,20 @@ const keplr = {
     return true;
   },
 
-  async getWalletAddress() {
-    await playwright.switchToKeplrWindow();
+  async getWalletAddress(chainName) {
+    playwright.switchToKeplrWindow();
     await module.exports.goToHome();
-    const page = playwright.keplrWindow();
+    const newTokensSelctorExists = await playwright.waitForAndCheckElementExistence(
+      homePageElements.newTokensFoundSelector,
+    );
+
+    if (newTokensSelctorExists) {
+      await module.exports.addNewTokensFound(false);
+    }
```

Comment:
dont think we need to do this internally. in most cases this will not be needed
in cases when it is needed, a user will know that tokens are not currently present and can call it themselves explicitly

Reply:
I don't think that is reliable. We should not assume that the user knows. A flow should be capable of handling all edge cases.

Reply:
this is not an edge case, the writer of a test should know if the wallet address is present or not before writing the test.

There can be cases where user does not want to press the 'add new token' button until a later test. and since our code is forcing him to add new tokens, it might ruin their flow.

Plus all our interactions should have a single fuctionality/responsibility. here we are performing two different function together which a user can always perform separately

Reply:
This is an edge case. 

Scenario:
The user want's agoric wallet address. Goes to copy the address but the agoric chain isn't present there ==> Mainly because the user did not include new tokens ==> Output is the user never got the address when they should've gotten it.

Also, assumptions lead to incorrect behaviour and unreliable software.  

As for SRP, I am all in for it. But then we should apply it consistently and where it's required.